Build.bee.cs

```diff
@@ -0,0 +1,377 @@
+                installDirectory.Combine("cmake/TBBConfig.cmake"),
+                installDirectory.Combine("cmake/TBBConfigVersion.cmake"),
+            },
+            inputs: new [] {buildDirectory, installDirectory.Combine("include/tbb/tbb_stddef.h")},
+            executableStringFor: "cmake",
+            commandLineArguments: new[] {
+                $"-DINSTALL_DIR={installDirectory.Combine("cmake").InQuotes(SlashMode.Native)}",
+                $"-DSYSTEM_NAME={SystemNameFor(programConfiguration.Sdk)}",
+                $"-DTBB_VERSION_FILE={installDirectory.Combine("include/tbb/tbb_stddef.h").InQuotes(SlashMode.Native)}",
+                "-DINC_REL_PATH=../include",
+                "-DLIB_REL_PATH=../lib",
+                "-DBIN_REL_PATH=../bin",
+                "-P cmake/tbb_config_installer.cmake"
+            },
+            environmentVariables: environmentVariables,
+            targetDirectories: new NPath[] {}
+        );
+    }
+
+    static string ArchFor(Architecture architecture)
+    {
+        if (architecture is x64Architecture) return "intel64";
+        if (architecture is Arm64Architecture) return "arm64";
+        throw new Exception($"Unsupported architecture {architecture} for {nameof(ArchFor)}");
+    }
+
+    static string CompilerFor(Sdk sdk)
+    {
+        if (sdk is VisualStudioSdk) return "cl";
+        if (sdk is ClangSdk || sdk is LinuxClangSdk) return "clang";
+        throw new Exception($"Unsupported Sdk {sdk} for {nameof(CompilerFor)}");
+    }
+
+    static string CfgFor(CodeGen codeGen)
+    {
+        if (codeGen is CodeGen.Release) return "release";
+        if (codeGen is CodeGen.Debug) return "debug";
+        throw new Exception($"Unsupported CodeGen {codeGen} for {nameof(CfgFor)}");
+    }
+
+    static string ChooseMinMacOSVersion(Architecture architecture)
+    {
+        if (architecture is Arm64Architecture)
+            return "11.0";
+
+        return "10.14";
+    }
+
+    static string SystemNameFor(Sdk sdk)
+    {
+        if (sdk is WindowsSdk) return "Windows";
+        if (sdk is MacSdk) return "Darwin";
+        if (sdk is LinuxClangSdk) return "Linux";
+        throw new Exception($"Unsupported Sdk {sdk} for {nameof(CompilerFor)}");
+    }
+
+    static void Install_Windows(NPath buildDirectory, NPath installDirectory, string targetId)
+    {
+        var binDirectory = installDirectory.Combine("bin");
+        var libDirectory = installDirectory.Combine("lib");
+        var includeDirectory = installDirectory.Combine("include");
+
+        Backend.Current.AddAction(actionName: "install",
+            targetFiles: new NPath[] {},
+            inputs: new[] {buildDirectory},
+            executableStringFor: "(robocopy"
+                + $" {buildDirectory.ToString(SlashMode.Native)}"
+                + $" {binDirectory.ToString(SlashMode.Native)}"
+                + " /s"
+                + " tbb*.dll"
+                + " tbb*.pdb"
+                + ") ^& EXIT /B 0",
```

Comment:
Robocopy exit codes are unusual (IIUC anything 0-8 means "success" and anything else means "failure"). Does this just ignore the robocopy exit code and always return 0? If so, that sounds bad.

I'm experimenting with a wrapper script in https://github.cds.internal.unity3d.com/unity/unity/pull/17421 - maybe something like that would work here?